### DISCUSSION TITLE
What´s the most suitable/recomended way to add commands and enviroments suggestions (autocomplete commands and enviroments)?

### DISCUSSION THREAD

**Discussion by rafa4mat** [WinEdt-Team/WinEdt-Support#75] [1.1 General Help and Discussion]
WinEdt 
- typing \fra...   offers me \frac and \frac{•}{•}
BUT
- typing \dfrac... offers me only \dfrac and NO \dfrac{•}{•}

What´s the best (most suitable/recomended) way to modify this issue?, for adding commands (\dfrac{•}{•} in this case).

Thinking in use this changes in future WinEdt upgrades.

**Comment by Alex-Simonic** [comment]
By modifying the `[AUTO_COMPLETE]` configuration section. If you send me a list of missing common commands, I will add them to the defaults. In this case, adding a new OBJECT based on \frac will do... Your changes are preserved during the upgrades (if you use the Options Interface) but you may miss any additions to the defaults (should anything change with this).

**Reply by rafa4mat** [reply]
Thank you, @Alex-Simonic . Works fine, after adding

```
OBJECT="\dfrac{'bullet'}{'bullet'}"
      MACRO="Exe('%b\Macros\Auto Complete\TeX_Arg.edt');"
```
at Options Interface / Delimiters, Auto Complete / AutoComplete

and secondary click on the upper tabs files over tab AutoComplete.ini , SAVE and then LOAD SCRIPT.

--------------------
Ok, If I had any other suggestions I´ll write to you.

Do you think are rigth suggestions include popular packages?,  as
-> tcolorbox (\begin{tcolorbox}[...]...)
-> mathtools (\underbracket{}_{})
-> wrapfigure2 (\begin{wrapfigure}{r}{lenght}...\centering..... )
...
